File: app/buildz/tmp/kapt3/stubs/debug/com/bits/bee/bpmc/data/repository/ProvinceRepositoryImpl.java
package com.bits.bee.bpmc.data.repository;

import java.lang.System;

@kotlin.Metadata(mv = {1, 7, 1}, k = 1, d1 = {"\u0000D\n\u0002\u0018\u0002\n\u0002\u0018\u0002\n\u0000\n\u0002\u0018\u0002\n\u0000\n\u0002\u0018\u0002\n\u0000\n\u0002\u0018\u0002\n\u0002\b\u0002\n\u0002\u0018\u0002\n\u0002\u0018\u0002\n\u0002\u0018\u0002\n\u0000\n\u0002\u0010\b\n\u0000\n\u0002\u0010\u0002\n\u0000\n\u0002\u0010 \n\u0002\u0018\u0002\n\u0002\b\u0002\u0018\u00002\u00020\u0001B\u001f\b\u0007\u0012\u0006\u0010\u0002\u001a\u00020\u0003\u0012\u0006\u0010\u0004\u001a\u00020\u0005\u0012\u0006\u0010\u0006\u001a\u00020\u0007\u00a2\u0006\u0002\u0010\bJ\u001c\u0010\t\u001a\u000e\u0012\n\u0012\b\u0012\u0004\u0012\u00020\f0\u000b0\n2\u0006\u0010\r\u001a\u00020\u000eH\u0016J\u001f\u0010\u000f\u001a\u00020\u00102\f\u0010\u0011\u001a\b\u0012\u0004\u0012\u00020\u00130\u0012H\u0096@\u00f8\u0001\u0000\u00a2\u0006\u0002\u0010\u0014R\u000e\u0010\u0002\u001a\u00020\u0003X\u0082\u0004\u00a2\u0006\u0002\n\u0000R\u000e\u0010\u0006\u001a\u00020\u0007X\u0082\u0004\u00a2\u0006\u0002\n\u0000R\u000e\u0010\u0004\u001a\u00020\u0005X\u0082\u0004\u00a2\u0006\u0002\n\u0000\u0082\u0002\u0004\n\u0002\b\u0019\u00a8\u0006\u0015"}, d2 = {"Lcom/bits/bee/bpmc/data/repository/ProvinceRepositoryImpl;", "Lcom/bits/bee/bpmc/domain/repository/ProvinceRepository;", "apiUtils", "Lcom/bits/bee/bpmc/data/data_source/remote/ApiUtils;", "provinceDao", "Lcom/bits/bee/bpmc/data/data_source/local/dao/ProvinceDao;", "defaultDispatcher", "Lkotlinx/coroutines/CoroutineDispatcher;", "(Lcom/bits/bee/bpmc/data/data_source/remote/ApiUtils;Lcom/bits/bee/bpmc/data/data_source/local/dao/ProvinceDao;Lkotlinx/coroutines/CoroutineDispatcher;)V", "getLatestProvince", "Lkotlinx/coroutines/flow/Flow;", "Lcom/bits/bee/bpmc/utils/Resource;", "Lcom/bits/bee/bpmc/data/data_source/remote/response/ProvinceResponse;", "page", "", "insertBulkProvince", "", "list", "", "Lcom/bits/bee/bpmc/data/data_source/local/model/ProvinceEntity;", "(Ljava/util/List;Lkotlin/coroutines/Continuation;)Ljava/lang/Object;", "app_debug"})
public final class ProvinceRepositoryImpl implements com.bits.bee.bpmc.domain.repository.ProvinceRepository {
    private final com.bits.bee.bpmc.data.data_source.remote.ApiUtils apiUtils = null;
    private final com.bits.bee.bpmc.data.data_source.local.dao.ProvinceDao provinceDao = null;
    private final kotlinx.coroutines.CoroutineDispatcher defaultDispatcher = null;
    
    @javax.inject.Inject()
    public ProvinceRepositoryImpl(@org.jetbrains.annotations.NotNull()
    com.bits.bee.bpmc.data.data_source.remote.ApiUtils apiUtils, @org.jetbrains.annotations.NotNull()
    com.bits.bee.bpmc.data.data_source.local.dao.ProvinceDao provinceDao, @org.jetbrains.annotations.NotNull()
    kotlinx.coroutines.CoroutineDispatcher defaultDispatcher) {
        super();
    }
    
    @org.jetbrains.annotations.Nullable()
    @java.lang.Override()
    public java.lang.Object insertBulkProvince(@org.jetbrains.annotations.NotNull()
    java.util.List<com.bits.bee.bpmc.data.data_source.local.model.ProvinceEntity> list, @org.jetbrains.annotations.NotNull()
    kotlin.coroutines.Continuation<? super kotlin.Unit> continuation) {
        return null;
    }
    
    @org.jetbrains.annotations.NotNull()
    @java.lang.Override()
    public kotlinx.coroutines.flow.Flow<com.bits.bee.bpmc.utils.Resource<com.bits.bee.bpmc.data.data_source.remote.response.ProvinceResponse>> getLatestProvince(int page) {
        return null;
    }
}

File: app/src/main/java/com/bits/bee/bpmc/data/repository/ProvinceRepositoryImpl.kt
package com.bits.bee.bpmc.data.repository

import com.bits.bee.bpmc.data.data_source.local.dao.ProvinceDao
import com.bits.bee.bpmc.data.data_source.local.model.ProvinceEntity
import com.bits.bee.bpmc.data.data_source.remote.ApiUtils
import com.bits.bee.bpmc.data.data_source.remote.response.BranchResponse
import com.bits.bee.bpmc.data.data_source.remote.response.ProvinceResponse
import com.bits.bee.bpmc.domain.mapper.ProvinceDataMapper
import com.bits.bee.bpmc.domain.model.Branch
import com.bits.bee.bpmc.domain.model.Province
import com.bits.bee.bpmc.domain.repository.ProvinceRepository
import com.bits.bee.bpmc.utils.ApiResponse
import com.bits.bee.bpmc.utils.NetworkBoundResource
import com.bits.bee.bpmc.utils.NetworkDatabaseBoundResource
import com.bits.bee.bpmc.utils.Resource
import kotlinx.coroutines.CoroutineDispatcher
import kotlinx.coroutines.flow.Flow
import kotlinx.coroutines.withContext
import javax.inject.Inject

class ProvinceRepositoryImpl @Inject constructor(
    private val apiUtils: ApiUtils,
    private val provinceDao: ProvinceDao,
    private val defaultDispatcher: CoroutineDispatcher
): ProvinceRepository {
    //    override fun getProvinceList(): Flow<Resource<List<Province>>> {
//        return object : NetworkDatabaseBoundResource<List<Province>, ProvinceResponse>(){
//            override suspend fun loadFormDB(): List<Province>? {
//                return provinceDao.getProvinceList().map { it.toProvince() }
//            }
//
//            override fun shouldFetch(data: List<Province>?): Boolean {
//                return true
//            }
//
//            override suspend fun createCall(): Flow<ApiResponse<ProvinceResponse>> {
//                return apiUtils.getProvinceApiService().getProvinceList()
//            }
//
//            override suspend fun saveCallResult(data: ProvinceResponse) {
//                provinceDao.insertBulk(data.data.data.map { ProvinceDataMapper.fromResponseToData(it) })
//            }
//
//        }.getAsFlow()
//    }
    override suspend fun insertBulkProvince(list: List<ProvinceEntity>) {
        withContext(defaultDispatcher){
            provinceDao.insertBulk(list)
        }
    }

    override fun getLatestProvince(page: Int): Flow<Resource<ProvinceResponse>> {
        return object : NetworkBoundResource<ProvinceResponse>(){
            override fun createCall(): Flow<ApiResponse<ProvinceResponse>> {
                return apiUtils.getProvinceApiService().getProvinceList(page)
            }

        }.getAsFlow()
    }
}